aioredis/commands/geo.py

```diff
@@ -112,44 +115,50 @@ def validate_georadius_options(radius, unit, count, sort_dir,
     return args
 
 
-def pairs_float(value):
-    return [[float(val[0]), float(val[1])] for val in value]
+def make_geo_coord(value):
+    return GeoCoord(*map(float, value))
+
 
+def make_geopos(value):
+    return [make_geo_coord(val) for val in value]
 
-def geo_data_row(value, with_dist, with_coord, with_hash):
+
+def make_geomember(member, distance, hash_, coord):
+    if distance is not None:
+        distance = float(distance)
+    if hash_ is not None:
+        hash_ = int(hash_)
+    if coord is not None:
+        coord = GeoCoord(*map(float, coord))
+
+    return GeoMember(member, distance, hash_, coord)
+
+
+def process_geomember(value, with_dist, with_coord, with_hash):
     res_rows = []
     for row in value:
-        res = []
-        
-        if with_dist and with_coord and with_hash:
-            res.append(row[0])
-            res.append(float(row[1]))
-            res.append(int(row[2]))
-
-            res.append([float(row[3][0]), float(row[3][1])])
-        elif with_dist and with_coord:
-            res.append(row[0])
-            res.append(float(row[1]))
-            res.append([float(row[2][0]), float(row[2][1])])
-        elif with_dist and with_hash:
-            res.append(row[0])
-            res.append(float(row[1]))
-            res.append(int(row[2]))
-        elif with_hash and with_coord:
-            res.append(row[0])
-            res.append(int(row[1]))
-            res.append([float(row[2][0]), float(row[2][1])])
-        elif with_dist:
-            res.append(row[0])
-            res.append(float(row[1]))
-        elif with_hash:
-            res.append(row[0])
-            res.append(int(row[1]))
-        elif with_coord:
-            res.append(row[0])
-            res.append([float(row[1][0]), float(row[1][1])])
+        member, distance, coord, hash_ = None, None, None, None
```

Comment:
Ok, my previous example was incorrect.
But I realy don't like this long if-elif noodles.
How about this:
```python
if isinstance(row, list):
    name = row.pop(0)
    dist = hash = coord = None
    if with_dist:
        dist = float(row.pop(0))
    if with_hash:
        hash = int(row.pop(0))
    if with_coord:
        coord = GeoCoord(*map(float, row.pop(0)))
    return GeoMember(name, dist, hash, coord)
```